x-pack/plugins/security_solution/public/common/components/tables/helpers.tsx

```diff
@@ -135,24 +156,116 @@ export const getRowItemDraggables = ({
-  rowItems: string[],
-  idPrefix: string,
+interface OverflowItemProps {
+  dataProvider?: DataProvider | DataProvider[] | undefined;
+  dragDisplayValue?: string;
+  field: string;
+  rowItem: string;
+}
+
+export const OverflowItemComponent: React.FC<OverflowItemProps> = ({
+  dataProvider,
+  dragDisplayValue,
+  field,
+  rowItem,
+}) => {
+  const [showTopN, setShowTopN] = useState<boolean>(false);
+  const { timelineId: timelineIdFind } = useContext(TimelineContext);
+  const [hoverActionsOwnFocus] = useState<boolean>(false);
+  const toggleTopN = useCallback(() => {
+    setShowTopN((prevShowTopN) => {
+      const newShowTopN = !prevShowTopN;
+      return newShowTopN;
+    });
+  }, []);
+
+  const closeTopN = useCallback(() => {
+    setShowTopN(false);
+  }, []);
+
+  return (
+    <EuiFlexGroup
+      gutterSize="none"
+      justifyContent="spaceBetween"
+      direction="row"
+      data-test-subj={`${field}-${dragDisplayValue ?? rowItem}`}
+    >
+      <EuiFlexItem grow={1}>{defaultToEmptyTag(rowItem)} </EuiFlexItem>
+      <EuiFlexItem grow={false} data-test-subj="hover-actions">
+        <HoverActions
+          closeTopN={closeTopN}
+          dataProvider={dataProvider}
+          field={field}
+          isObjectArray={false}
+          ownFocus={hoverActionsOwnFocus}
+          showOwnFocus={false}
+          showTopN={showTopN}
+          timelineId={timelineIdFind}
+          toggleTopN={toggleTopN}
+          values={dragDisplayValue ?? rowItem}
+        />
+      </EuiFlexItem>
+    </EuiFlexGroup>
+  );
+};
+
+OverflowItemComponent.displayName = 'OverflowItemComponent';
+export const OverflowItem = React.memo(OverflowItemComponent);
+
+interface RowItemOverflowProps {
+  attrName: string;
+  dragDisplayValue?: string;
+  idPrefix: string;
+  maxOverflowItems: number;
+  overflowIndexStart: number;
+  rowItems: string[];
+}
+
+export const RowItemOverflowComponent: React.FC<RowItemOverflowProps> = ({
+  attrName,
+  dragDisplayValue,
+  idPrefix,
+  maxOverflowItems = 5,
   overflowIndexStart = 5,
-  maxOverflowItems = 5
-): JSX.Element => {
+  rowItems,
+}) => {
+  const overflowItems = useMemo(
+    () =>
+      rowItems
+        .slice(overflowIndexStart, overflowIndexStart + maxOverflowItems)
+        .map((rowItem, index) => {
+          const id = escapeDataProviderId(`${idPrefix}-${attrName}-${rowItem}-${index}`);
+          const dataProvider = {
+            and: [],
+            enabled: true,
+            id,
+            name: rowItem,
+            excluded: false,
+            kqlQuery: '',
+            queryMatch: {
+              field: attrName,
+              value: rowItem,
+              displayValue: dragDisplayValue || rowItem,
+              operator: IS_OPERATOR as QueryOperator,
+            },
+          };
+
+          return (
+            <EuiFlexItem key={`${idPrefix}-${id}`}>
+              <OverflowItem dataProvider={dataProvider} rowItem={rowItem} field={attrName} />
```

Comment:
`OverflowItem` has `dragDisplayValue` property, but I don't see it being used. Should you pass it down here?